src/components/Copy/index.tsx

```diff
@@ -0,0 +1,154 @@
+import { faCopy } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Box, Tooltip, useTheme } from '@mui/material';
+import React, { useEffect, useState } from 'react';
+
+export interface CopyProps {
+	copy: string | undefined;
+	onClick?: () => void;
+}
+
+export interface CopyPropsWithChildren extends CopyProps {
+	children: React.ReactNode;
+	variant?: 'absolute' | 'adjacent';
+}
+
+export const copyToClipboard = async (text: string): Promise<void> => {
+	try {
+		await navigator.clipboard.writeText(text);
+	} catch (err) {
+		console.error('Failed to copy text: ', err);
+	}
+};
+
+export const Copy = ({
+	copy,
+	onClick,
+	...props
+}: CopyProps | CopyPropsWithChildren) => {
+	let closeTooltipTimeout: ReturnType<typeof setInterval> | null = null;
+	const [open, setOpen] = useState(false);
+	const theme = useTheme();
+
+	useEffect(() => {
+		if (!closeTooltipTimeout) {
+			return;
+		}
+		return clearTimeout(closeTooltipTimeout);
+	}, [closeTooltipTimeout]);
+
+	const handleTooltipOpen = () => {
+		setOpen(true);
+		closeTooltipTimeout = setTimeout(() => setOpen(false), 1000);
+	};
+
+	const copyClick = React.useCallback(
+		(e: React.MouseEvent) => {
+			e.preventDefault();
+			e.stopPropagation();
+			handleTooltipOpen();
+			onClick?.();
+			// TODO: Just to make TS happy. This will never happen because we return null if there is no copy prop
+			if (copy) {
+				copyToClipboard(copy);
+			}
```

Comment:
let's just do 

if (!copy) {
 return; 
}

on top so we also avoid handleTooltipOpen and onClick for consistency